workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityComponentDataGenerator.tt

```diff
@@ -19,7 +20,33 @@ namespace <#= qualifiedNamespace #>
         {
             public uint ComponentId => <#= componentDetails.ComponentId #>;
 
-            public BlittableBool DirtyBit { get; set; }
+            [Flags]
+            public enum DirtyBitsEnum
+            {
+                None = 0,
+<# for(int i = 0; i < fieldDetailsList.Count; i++) { #>
+                <#= fieldDetailsList[i].PascalCaseName #> = <#= 1 << i #>,
+<# } #>
+            }
+
+            private DirtyBitsEnum dirtyBits;
+
+            public DirtyBitsEnum DirtyBits => dirtyBits;
+
+            public bool IsDirty()
+            {
+                return (DirtyBits != DirtyBitsEnum.None);
+            }
+
+            public bool IsDirty(DirtyBitsEnum currentField)
+            {
+                return (dirtyBits & currentField) != DirtyBitsEnum.None;
+            }
+
+            public void ClearDirtyBits()
```

Comment:
We don't actually need to specify that we are doing this with a bit mask in the name. The API is that you mark something as no dirty. Bits are just an implementation that the caller doesn't need to know.

Reply:
MarkNotDirty would be better

Reply:
Done!